apps/angular/projection/src/app/ui/list-item/list-item.component.ts

```diff
@@ -17,19 +14,9 @@ import { CardType } from '../../model/card.model';
 })
 export class ListItemComponent {
   @Input() id!: number;
-  @Input() name!: string;
-  @Input() type!: CardType;
+  @Output() listItemDelete = new EventEmitter();
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
-
-  delete(id: number) {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.deleteOne(id);
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.deleteOne(id);
-    }
+  deleteId() {
+    this.listItemDelete.emit(this.id);
```

Comment:
here you don't need the id. this can be a simple event Emitter that trigger the delete action and you pass the id in the parent.